controllers/rediscluster_controller.go

```diff
@@ -61,7 +61,7 @@ func (r *RedisClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-	if instance.Spec.RedisLeader.Replicas != nil && *instance.Spec.RedisLeader.Replicas != 0 {
+	if (instance.Spec.Size != nil && *instance.Spec.Size != 0) || (instance.Spec.RedisLeader.Replicas != nil && *instance.Spec.RedisLeader.Replicas != 0) {
```

Comment:
If we only used the `instance.Spec.Size` field, we never deployed a service, even though we were deploying equal Follower/Leaders.